theano/gradient.py

```diff
@@ -1669,7 +1675,11 @@ def random_projection():
     # This sum() is defined above, it's not the builtin sum.
     cost = theano.tensor.sum(t_r * o_output)
 
-    cost_fn = function(tensor_pt, cost, name='gradient.py cost')
+    if no_debug_ref:
+        cost_fn = function(tensor_pt, cost, name='gradient.py cost',
+                           mode=mode_not_debug(mode))
+    else:
+        cost_fn = function(tensor_pt, cost, name='gradient.py cost')
```

Comment:
This version of the code is more clear:
```
mode_for_cost=mode_not_debug(mode)
cost_fn = function(tensor_pt, cost, name..., mode=mode_for_cost)
```

We see clearly that the only different is the mode.